File: Plugins/AM_PRIDE_Converter_PlugIn/clsPXFileInfo.vb
Imports AnalysisManagerBase.clsAnalysisResources

Public Class clsPXFileInfo
    Inherits clsPXFileInfoBase

#Region "Module Variables"
    Protected mFileMappings As List(Of Integer)
#End Region

#Region "Auto-properties"
    Public Property PXFileType As ePXFileType
#End Region

#Region "Properties"
    Public ReadOnly Property FileMappings() As List(Of Integer)
        Get
            Return mFileMappings
        End Get
    End Property
#End Region

    Public Sub New(fileName As String, udtJobInfo As udtDataPackageJobInfoType)
        MyBase.New(fileName, udtJobInfo)
        mFileMappings = New List(Of Integer)
    End Sub

    Public Sub AddFileMapping(intPXFileID As Integer)
        If Not mFileMappings.Contains(intPXFileID) Then
            mFileMappings.Add(intPXFileID)
        End If
    End Sub

End Class

File: Plugins/AM_PRIDE_Converter_PlugIn/clsPXFileInfo.cs
using AnalysisManagerBase;
using System.Collections.Generic;

namespace AnalysisManagerPRIDEConverterPlugIn
{
    /// <summary>
    /// ProteomeXchange file info
    /// </summary>
    public class clsPXFileInfo : clsPXFileInfoBase
    {
        // Ignore Spelling: ProteomeXchange

        #region "Properties"

        /// <summary>
        /// ProteomeXchange file type
        /// </summary>
        public ePXFileType PXFileType { get; set; }

        /// <summary>
        /// Mapping from this file to parent files
        /// </summary>
        public List<int> FileMappings { get; }

        #endregion

        /// <summary>
        /// Constructor
        /// </summary>
        /// <param name="fileName"></param>
        /// <param name="dataPkgJob"></param>
        public clsPXFileInfo(string fileName, clsDataPackageJobInfo dataPkgJob) : base(fileName, dataPkgJob)
        {
            FileMappings = new List<int>();
        }

        /// <summary>
        /// Add a file mapping
        /// </summary>
        /// <param name="parentFileId">Parent FileID</param>
        public void AddFileMapping(int parentFileId)
        {
            if (!FileMappings.Contains(parentFileId))
            {
                FileMappings.Add(parentFileId);
            }
        }
    }
}
